js/index.js

```diff
@@ -0,0 +1,632 @@
+            event.layerX > playerDrawCoord[active].x + 290 &&
+            event.layerX <
+              playerDrawCoord[active].x + 290 + (track + 1) * (50 + 8) &&
+            event.layerY > playerDrawCoord[active].y + track * (50 + 8) &&
+            event.layerY < playerDrawCoord[active].y + (track + 1) * (50 + 8) &&
+            this.selectedTile.color !== 0 &&
+            this.players[active].storage[track].available === true
+          ) {
+            console.log(`Track #${track} has been clicked.`);
+            for (let i = 0; i < 5; i++) {
+              this.players[active].storage[i].available = false;
+            }
+            if (this.selectedTile.firstMarker === true) {
+              this.players[active].penaltyLine.push(0);
+              this.selectedTile.firstMarker = false;
+            }
+            let availableSlots =
+              this.players[active].storage[track].maxSlots -
+              this.players[active].storage[track].usedSlots;
+            if (this.selectedTile.amount <= availableSlots) {
+              this.players[active].storage[
+                track
+              ].usedSlots += this.selectedTile.amount;
+            } else {
+              this.selectedTile.amount -= availableSlots;
+              this.players[active].storage[track].usedSlots = this.players[
+                active
+              ].storage[track].maxSlots;
+              for (
+                let extraTiles = 0;
+                extraTiles < this.selectedTile.amount;
+                extraTiles++
+              ) {
+                this.players[active].penaltyLine.push(this.selectedTile.color);
+              }
+            }
+            this.players[active].storage[
+              track
+            ].tileID = this.selectedTile.color;
+
+            this.emptyFactory();
+            this.switchPlayer();
+          }
+        }
+      }
+    });
+  }
+  emptyFactory() {
+    if (this.selectedTile.factory !== undefined) {
+      for (let tile = 0; tile < 4; tile++) {
+        if (
+          this.factoryTileSelected[this.selectedTile.factory][tile] === false
+        ) {
+          this.discardYard.push(
+            this.factories[this.selectedTile.factory][tile]
+          );
+        }
+      }
+      this.factories[this.selectedTile.factory] = [];
+    } else {
+      for (let i = this.discardYard.length; i >= 0; i--) {
+        if (this.discardSelected[i] === 1) {
+          this.discardYard.splice(i, 1);
+          this.discardSelected.splice(i, 1);
+        }
+      }
+    }
+
+    this.selectedTile.factory = undefined;
+    this.selectedTile.color = 0;
+    this.selectedTile.amount = 0;
+    this.selectedTile.firstMarker = false;
+    this.factoryTileSelected[this.selectedTile.factory] = [
+      false,
+      false,
+      false,
+      false
+    ];
+  }
+  switchPlayer() {
+    if (this.currentPlayer === this.playerCount - 1) {
+      this.currentPlayer = 0;
+    } else {
+      this.currentPlayer++;
+    }
```

Comment:
Instead, could be something like `this.currentPlayer = (this.currentPlayer + 1) % this.playerCount;`